code/game/machinery/computer/Operating.dm

```diff
@@ -95,7 +94,28 @@
 		usr.set_machine(src)
 	return
 
-
 /obj/machinery/computer/operating/process()
 	if(..())
 		src.updateDialog()
+	update_icon()
+
+/obj/machinery/computer/operating/update_icon()
+	..()
+	icon_state = initial(icon_state)
+	// Broken
+	if(stat & BROKEN)
+		icon_state += "b"

```

Comment:
You really shouldn't build icon_states like this, they're a lot less safe than regular strings.